client-mr/src/main/java/org/apache/hadoop/mapred/SortWriteBufferManager.java

```diff
@@ -312,7 +312,7 @@ ShuffleBlockInfo createShuffleBlock(SortWriteBuffer wb) {
     final byte[] compressed = RssShuffleUtils.compressData(data);
     final long crc32 = ChecksumUtils.getCrc32(compressed);
     compressTime += System.currentTimeMillis() - start;
-    final long blockId = RssMRUtils.getBlockId(partitionId, taskAttemptId, getNextSeqNo(partitionId));
+    final long blockId = RssMRUtils.getBlockId(Long.valueOf(partitionId), taskAttemptId, getNextSeqNo(partitionId));
```

Comment:
`Long.valueof(partitionId)` -> (long)partitionId

Reply:
Reduce the cost of box.

Reply:
Actually，we can directly remove the type conversion, use of implicit type casting : Long.valueof(partitionId) ->  partitionId

Reply:
> Is it necessary to change this line? Pass int value to long argument should be fine.

Just tell the reader of code that we're sure that we need this type cast.